packages/ipfs-cpinner-provider/src/IpfsClient.ts

```diff
@@ -0,0 +1,25 @@
+import { IpfsHttpClient, IpfsObject } from 'ipfs-http-client'
+import { Content, CID, IpfsClient } from './types'
+import all from 'it-all'
+
+export default class implements IpfsClient {
+  constructor (private ipfsHttpClient: IpfsHttpClient) {}
+
+  async get (cid: string): Promise<string[]> {
+    const contents = await all(this.ipfsHttpClient.get(cid))
```

Comment:
`Promise.all` takes an array of promises... why do we need this dependency?

Reply:
Because the `ipfs-http-client` returns an `AsyncGenerator`. That dependency iterates the entire generator. Took the example from https://github.com/ipfs/js-ipfs/blob/master/packages/ipfs-http-client/test/get.spec.js#L34-L40

Reply:
It does `map` inside https://github.com/achingbrain/it/blob/master/packages/it-all/index.js